inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/sidebar/RecommendationSidebar.java

```diff
@@ -17,55 +17,241 @@
+
+				// avoids a ConcurrentModificationException
+				it.remove();
+			}
+
+			String javascript = createJSScript(dataColumns.toString(), chartType.toString());
+			log.info(javascript);
+
+			aEvent.getRequestHandler().prependJavaScript(javascript);
+		} catch (IOException e) {
+			log.error("Unable to render chart", e);
+			error("Unable to render chart: " + e.getMessage());
+			aEvent.getRequestHandler().addChildren(getPage(), IFeedback.class);
+		}
+	}
+
+	/**
+	 * Creates the js script to render graph with the help of given data points. Sample value of dataColumns:
+	 * ['recommender1', 1.0, 2.0, 3.0 ], ['recommender2', 2.0, 3.0, 4.0]. Also creates an xaxix of a sequence
+	 * from 0 to maximumNumberOfPoints (50)
+	 * 
+	 * @param dataColumns
+	 * @param chartType 
+	 * @return
+	 * @throws IOException
+	 */
+	private String createJSScript(String dataColumns, String chartType) throws IOException {
+		int[] intArray = IntStream.range(0, maxPointsToPlot).map(i -> i).toArray();
+
+		String xaxisValues = "[ 'x' ," + substring(Arrays.toString(intArray), 1, -1) + "]";
+		String data = toJsonString(dataColumns).substring(1, dataColumns.toString().length());
+
+		
+		// bind data to chart container
+		String javascript = "var chart=c3.generate({bindto:'#" + chartContainer.getMarkupId()
+				+ "',data:{ x:'x', columns:[" + xaxisValues + " ," + data + "],types:{"+chartType +"}}});;";
+		return javascript;
+	}
+
+	/**
+	 * Fetches a number of latest evaluation scores from the database and save it in
+	 * the map corresponding to each recommender for which the scores have been
+	 * logged in the database
+	 */
+	private HashMap<String, List<Double>> getLatestScores(RenderAnnotationsEvent aEvent) {
+		// we want to plot RecommenderEvaluationResultEvent for the learning curve. The
+		// value of the event
+		String eventType = "RecommenderEvaluationResultEvent";
+
+		List<LoggedEvent> loggedEvents = eventRepo.listLoggedEvents(aModel.getObject().getProject(),
+				aModel.getObject().getUser().getUsername(), eventType, maxPointsToPlot);
+
+		//we want to show the latest record on the right side of the graph
+		Collections.reverse(loggedEvents);
+
+		HashMap<String, List<Double>> recommenderScoreMap = new HashMap<>();
+
+		// iterate over the logged events to extract the scores and map
+		// it against its corresponding recommender.
+		for (LoggedEvent loggedEvent : loggedEvents) {
+			String detailJson = loggedEvent.getDetails();
+			try {
+				Details detail = fromJsonString(Details.class, detailJson);
+				List<Double> list = recommenderScoreMap.get(detail.tool);
+				if (list==null) {
+					list = new ArrayList<Double>();
+
+					addScoreToMap(recommenderScoreMap, detail.score, detail.tool, list);
+					continue;
+				}
+
+				addScoreToMap(recommenderScoreMap, detail.score, detail.tool, list);
+			
+			} catch (IOException e) {
+				log.error("Invalid logged Event detail. Skipping record with logged event id: " + loggedEvent.getId(),
```

Comment:
You can use String formatting for arguments in logging, e.g.
`log.error("Invalid logged Event detail. Skipping record with logged event id [{}]", loggedEvent.getId(), e);`
The `{}` are replaced with the arguments in the vararg, the stacktrace of the exception is logged if it is last, it does not need a placeholder.